modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java

```diff
@@ -845,6 +845,11 @@ public String toAllOfName(List<String> names, ComposedSchema composedSchema) {
             }
             return schemaType;
         }).distinct().collect(Collectors.toList());
-        return String.join(" & ", types);
+
+        if (types.size() > 1 && types.contains("any")) {
```

Comment:
nit: it might be simpler to add `.filter()` call to the stream instead of post-processing the resulting list